index.js

```diff
@@ -1,162 +1,152 @@
+              'pageTitle': document.title,
+              'mimeType': mimeType,
+              'duration': getMediaDurationInSeconds(target),
+              'detected': true
+            })
           }
 
           if (node.closest('audio') === target) {
-            tagNode(target);
             sources.push({
-              "name": name,
-              "src": node.src,
-              "pageSrc": window.location.href,
-              "pageTitle": document.title,
-              "mimeType": mimeType,
-              "duration": clamp_duration(target.duration),
-              "detected": true,
-              "tagId": target.tagUUID,
-              thumbnail
-            });
+              'name': name,
+              'src': fixUpRelativeUrl(node.src),
+              'pageSrc': window.location.href,
+              'pageTitle': document.title,
+              'mimeType': mimeType,
+              'duration': getMediaDurationInSeconds(target),
+              'detected': true
+            })
           }
         }
-        
-      });
-      return sources;
+      })
+      return sources
     }
   }
 
-  function getNodeData(node, thumbnail) {
-    return getNodeSource(node, node.src, node.type, thumbnail);
+  function getAllVideoElements () {
+    return document.querySelectorAll('video')
   }
 
-  function getAllVideoElements() {
-    return document.querySelectorAll('video');
+  function getAllAudioElements () {
+    return document.querySelectorAll('audio')
   }
 
-  function getAllAudioElements() {
-    return document.querySelectorAll('audio');
+  function getThumbnail () {
+    const isThumbnailValid = (thumbnail) => { return thumbnail && thumbnail !== '' }
+
+    let thumbnail = document.querySelector('meta[property="og:image"]')?.content
+    return fixUpRelativeUrl(thumbnail)
   }
 
-  function getOGTagImage() {
-    return document.querySelector('meta[property="og:image"]')?.content
+  function getMediaTitle (node) {
+    const isTitleValid = (title) => { return title && title !== '' }
+
+    let title = node.title
+    if (!isTitleValid(title)) { title = document.title }
+
+    return title
+  }
+
+  function getMediaAuthor (node) {
+    // TODO(sko) Get metadata of author
+    return null
   }
 
-  let videoElements = getAllVideoElements() ?? [];
-  let audioElements = getAllAudioElements() ?? [];
-  const thumbnail = getOGTagImage();
+  function getMediaDurationInSeconds (node) {
+    let duration = node.duration
+    const isNan = (value) => { return typeof value === 'number' && Number.isNaN(value) }
+    const isInfinite = (value) => { return typeof value === 'number' && (value === Infinity || value === -Infinity) }
+    const clampDuration = (value) => {
+      if (isNan(value)) {
+        return 0.0
+      }
+
+      if (isInfinite(value)) {
+        return Number.MAX_VALUE
+      }
+      return value
+    }
+
+    return clampDuration(duration)
+  }
+
+  let videoElements = getAllVideoElements() ?? []
```

Comment:
Optional nit, as we don't reassign: 
```suggestion
  const videoElements = getAllVideoElements() ?? []
```